buildSrc/src/main/java/org/elasticsearch/gradle/ElasticsearchJavaPlugin.java

```diff
@@ -0,0 +1,378 @@
+            System.getProperties().entrySet().forEach(entry -> {
+                if ((entry.getKey().toString().startsWith("tests.") || entry.getKey().toString().startsWith("es."))) {
+                    test.systemProperty(entry.getKey().toString(), entry.getValue());
+                }
+            });
+
+            // TODO: remove this once ctx isn't added to update script params in 7.0
+            test.systemProperty("es.scripting.update.ctx_in_params", "false");
+
+            // TODO: remove this property in 8.0
+            test.systemProperty("es.search.rewrite_sort", "true");
+
+            // TODO: remove this once cname is prepended to transport.publish_address by default in 8.0
+            test.systemProperty("es.transport.cname_in_publish_address", "true");
+
+            // Set netty system properties to the properties we configure in jvm.options
+            test.systemProperty("io.netty.noUnsafe", "true");
+            test.systemProperty("io.netty.noKeySetOptimization", "true");
+            test.systemProperty("io.netty.recycler.maxCapacityPerThread", "0");
+
+            test.testLogging(logging -> {
+                logging.setShowExceptions(true);
+                logging.setShowCauses(true);
+                logging.setExceptionFormat("full");
+            });
+
+            if (OS.current().equals(OS.WINDOWS) && System.getProperty("tests.timeoutSuite") == null) {
+                // override the suite timeout to 30 mins for windows, because it has the most inefficient filesystem known to man
+                test.systemProperty("tests.timeoutSuite", "1800000!");
+            }
+
+            /*
+             *  If this project builds a shadow JAR than any unit tests should test against that artifact instead of
+             *  compiled class output and dependency jars. This better emulates the runtime environment of consumers.
+             */
+            project.getPluginManager().withPlugin("com.github.johnrengelman.shadow", p -> {
+                // Remove output class files and any other dependencies from the test classpath, since the shadow JAR includes these
+                FileCollection mainRuntime = project.getExtensions()
+                    .getByType(SourceSetContainer.class)
+                    .getByName(SourceSet.MAIN_SOURCE_SET_NAME)
+                    .getRuntimeClasspath();
+                // Add any "shadow" dependencies. These are dependencies that are *not* bundled into the shadow JAR
+                Configuration shadowConfig = project.getConfigurations().getByName(ShadowBasePlugin.getCONFIGURATION_NAME());
+                // Add the shadow JAR artifact itself
+                FileCollection shadowJar = project.files(project.getTasks().named("shadowJar"));
+
+                test.setClasspath(test.getClasspath().minus(mainRuntime).plus(shadowConfig).plus(shadowJar));
+            });
+        });
+    }
+
+    private static boolean inFipsJvm() {
```

Comment:
Let's ditch this and just use `BuildParams.inFipsJvm()`. No need to have this implemented twice.